### PR TITLE
SES cu main table story book item row

### PR DESCRIPTION
**Pull request by alejandrocabriales**
# Ticket
https://trello.com/c/vl1Dc2j6/222-feature-various-ux-improvements

# Actions
Add Item List

### CODE REVIEW

File: src/stories/components/custom-table/custom-table.stories.tsx [hunk 1]
```diff
@@ -1,36 +1,95 @@
...
+        value: CoreUnit,
+        key: CoreUnit.shortCode,
+      },
+      {
+        value: CoreUnit,
+        key: CoreUnit.shortCode,
+      },
+      {
+        value: CoreUnit,
+        key: CoreUnit.shortCode,
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/custom-table.stories.tsx:50]: Note that all those keys have the same value, they should be different

File: src/stories/components/custom-table/custom-table.stories.tsx [hunk 2]
```diff
@@ -1,36 +1,95 @@
...
+      },
+    ] as CustomTableRow[],
+    loading: false,
+    sortState: [] as SortEnum[],
+    handleSort: () => undefined,
+    headersSort,
+    // renderCard: (data: CustomTableRow, index: number) =>,
+    // renderCard?: (data: CustomTableRow, index: number) => JSX.Element //(data: CustomTableRow, index: number) => JSX.Element;
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/custom-table.stories.tsx:58]: are those comments required?

File: src/stories/components/custom-table/custom-table.stories.tsx [hunk 3]
```diff
@@ -1,36 +1,95 @@
...
+        component:
+          'https://www.figma.com/file/pyaYEjcwF2b5uf9y0vIfIy/SES-Dashboard?node-id=8868%3A139602&t=BJYSrpHcrXx2uY02-4',
+        options: {
+          style: {
+            top: '-2px',
+            left: '30px',
+            // width: '100%',
+            // display: 'flex',
+            // flexGrow: 'row',
+            // justifyContent: 'center',
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/custom-table.stories.tsx:80]: please, remove those comments if they're no required anymore

File: src/stories/components/custom-table/custom-table.stories.tsx [hunk 4]
```diff
@@ -1,36 +1,95 @@
...
-  headers: ['Lorem', 'Ipsum', 'Dolor', 'Sit', 'Amet'],
-  items: [],
+Table.parameters = {
+  figma: {
+    component: {
+      375: 'https://www.figma.com/file/pyaYEjcwF2b5uf9y0vIfIy/SES-Dashboard?node-id=8868%3A143329&t=t4FrzPIVoc7xPgpL-4',
+      834: 'https://www.figma.com/file/pyaYEjcwF2b5uf9y0vIfIy/SES-Dashboard?node-id=5232%3A93999&t=t4FrzPIVoc7xPgpL-4',
+      1194: {
+        component:
+          'https://www.figma.com/file/pyaYEjcwF2b5uf9y0vIfIy/SES-Dashboard?node-id=8868%3A139602&t=BJYSrpHcrXx2uY02-4',
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/custom-table.stories.tsx:72]: These breakpoints should include a component style to fix the width according to the image width so the image match perfectly

File: src/stories/components/custom-table/list-core-unit/list-core-unit.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+import ListCoreUnit from './list-core-unit';
+import type { ComponentMeta } from '@storybook/react';
+
+export default {
+  title: 'Components/CUTable/ListCoreUnit',
+  component: ListCoreUnit,
+  parameters: {
+    chromatic: {
+      viewports: [1194, 1280, 1440],
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/list-core-unit/list-core-unit.stories.tsx:12]: I don't see any difference in the component UI in the breakpoints 1194 and 1280, so there should be one of them only

File: src/stories/components/custom-table/row-item/item-core-unit.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  decorators: [
+    (Story) => (
+      <div
+        style={{
+          margin: '3em',
+          width: 'fit-content',
+        }}
+      >
+        <Story />
+      </div>
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/row-item/item-core-unit.stories.tsx:25]: This decorator is unnecessary here as it only adds more difficulty to see, match with the breakpoint widths, and match with the Figma image.

File: src/stories/components/custom-table/row-item/item-core-unit.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    ),
+  ],
+} as ComponentMeta<typeof ItemCoreUnit>;
+
+const variantsArgs = [
+  {
+    queryStrings: '',
+    key: '2',
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/row-item/item-core-unit.stories.tsx:33]: the key should be job of the component, specifically inside the loop based on the CU data

File: src/stories/components/custom-table/row-item/item-core-unit.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    loading: false,
+    columns,
+    cu: CoreUnit,
+  },
+];
+export const [[Summary, SummaryDark]] = createThemeModeVariants(ItemCoreUnit, variantsArgs);
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/row-item/item-core-unit.stories.tsx:39]: Please, use other names according to the component name, as this is not related with summary and can be confusing

File: src/stories/components/custom-table/list-core-unit/list-core-unit.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        key: `${CoreUnit.code}+2`,
+      },
+      {
+        value: CoreUnit,
+        key: `${CoreUnit.code}+3`,
+      },
+    ],
+  },
+];
+export const [[Summary, SummaryDark]] = createThemeModeVariants(ListCoreUnit, variantsArgs);
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/list-core-unit/list-core-unit.stories.tsx:43]: please, use other names here as this is not related with summary

**Review by YasielCabrera — CHANGES_REQUESTED** [review]
I see you're not fixing the story component width when the component is different in several breakpoints, in those cases it is necessary to fix the width, otherwise the component will be bigger that the image and some elements are not going to match with the image. Look at this example:

```js
figma: {
    component: '...',
      834: {
        component: 'https://figma.com/...',
        options: {
          componentStyle: {
            // these styles are applied to the Story wrapper
            width: 834, // it may be other width depending on the image
          },
          style: {
            // these styles are applied to the figma image wrapper
            top: -35,
            left: -42,
          },
        },
      },
}
// ...
```

Another important thing I see is related to the Core Unit items and the Core Unit List, following a little bit the OOP paradigms, I think it is the responsibility of a CoreUnitItem how to be displayed in different breakpoints, I mention this because we have two different components, one when for small breakpoints that behave as a card and another that is like a table row, this should be handled by the same component instead of having different components and delegating the render responsibility to other components. Later the CoreUnitList should be able to render the list more easily, this way the CuTable component will have less login and all the related components will be easier to maintain.

File: src/stories/components/custom-table/custom-table.stories.tsx (hunk 3 above)
Comment (alejandrocabriales) [reply src/stories/components/custom-table/custom-table.stories.tsx:80]: ok, done

File: src/stories/components/custom-table/custom-table.stories.tsx (hunk 2 above)
Comment (alejandrocabriales) [reply src/stories/components/custom-table/custom-table.stories.tsx:58]: no my mistake, removed

File: src/stories/components/core-unit-card/core-unit-item-card.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  },
+} as ComponentMeta<typeof CoreUnitCard>;
+
+const variantsArgs = [
+  {
+    isLoading: false,
+    coreUnit: CoreUnit,
+  },
+];
+export const [[Summary, SummaryDark]] = createThemeModeVariants(CoreUnitCard, variantsArgs);
```
[the remark this answers is not in the document]
Comment (alejandrocabriales) [reply src/stories/components/core-unit-card/core-unit-item-card.stories.tsx:24]: ok

File: src/stories/components/custom-table/row-item/item-core-unit.stories.tsx (hunk 8 above)
Comment (alejandrocabriales) [reply src/stories/components/custom-table/row-item/item-core-unit.stories.tsx:39]: done

File: src/stories/components/custom-table/row-item/item-core-unit.stories.tsx (hunk 7 above)
Comment (alejandrocabriales) [reply src/stories/components/custom-table/row-item/item-core-unit.stories.tsx:33]: you are right

File: src/stories/components/custom-table/custom-table-mobile/card-mobile.tsx [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+interface Props {
+  coreUnit: CoreUnitDto;
+  keyForSkeleton?: number;
+}
+
+const CardItemCoreUnitMobile = ({ coreUnit, keyForSkeleton }: Props) => {
+  if (!coreUnit) {
+    return <CoreUnitCard coreUnit={{} as CoreUnitDto} key={`card-placeholder-${keyForSkeleton}`} isLoading />;
+  }
+  return <CoreUnitCard coreUnit={coreUnit as CoreUnitDto} key={`card-${coreUnit?.code}`} />;
```
[the remark this answers is not in the document]
Comment (alejandrocabriales) [reply src/stories/components/custom-table/custom-table-mobile/card-mobile.tsx:14]: not, you are right

File: src/stories/components/custom-table/row-item/item-core-unit.stories.tsx (hunk 6 above)
Comment (alejandrocabriales) [reply src/stories/components/custom-table/row-item/item-core-unit.stories.tsx:25]: done

File: src/stories/components/custom-table/list-core-unit/list-core-unit.stories.tsx (hunk 9 above)
Comment (alejandrocabriales) [reply src/stories/components/custom-table/list-core-unit/list-core-unit.stories.tsx:43]: done

File: src/stories/components/custom-table/custom-table.stories.tsx (hunk 1 above)
Comment (alejandrocabriales) [reply src/stories/components/custom-table/custom-table.stories.tsx:50]: done, the key is not necessary is manager by the component

File: src/stories/components/core-unit-card/core-unit-item-card.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+Summary.parameters = {
+  figma: {
+    component: {
+      375: 'https://www.figma.com/file/pyaYEjcwF2b5uf9y0vIfIy/SES-Dashboard?node-id=8868%3A143329&t=Hc8aw66dT23Z4EPe-4',
+      834: 'https://www.figma.com/file/pyaYEjcwF2b5uf9y0vIfIy/SES-Dashboard?node-id=8946%3A138834&t=Hc8aw66dT23Z4EPe-4',
```
[the remark this answers is not in the document]
Comment (alejandrocabriales) [reply src/stories/components/core-unit-card/core-unit-item-card.stories.tsx:30]: done

File: src/stories/components/custom-table/row-item/item-core-unit.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+import { columns, CoreUnit } from '@ses/core/utils/test.utils';
+import ItemCoreUnit from './item-core-unit';
+import type { ComponentMeta } from '@storybook/react';
+
+export default {
+  title: 'Components/CUTable/ItemCoreUnit',
+  component: ItemCoreUnit,
+  parameters: {
+    chromatic: {
+      viewports: [375, 834, 1194],
```
Comment (YasielCabrera) [inline src/stories/components/custom-table/row-item/item-core-unit.stories.tsx:11]: you should include 1440 and 1920 too as this component has has a different look and feel in those breakpoints

File: src/stories/containers/cu-table/cu-table.renders.tsx [hunk 14]
```diff
@@ -121,3 +101,37 @@ export const renderLastModified = (coreUnit: CoreUnitDto) => {
...
+  paddingRight: '16px',
+  width: '100%',
+  height: '50px',
+  margin: 'auto 0',
+  cursor: 'pointer',
+  [lightTheme.breakpoints.up('desktop_1194')]: {
+    paddingRight: '0px',
+    height: 'fit-content',
+  },
+  '@media (min-width: 1440px)': {
```
Comment (YasielCabrera) [inline src/stories/containers/cu-table/cu-table.renders.tsx:118]: You should use `[lightTheme.breakpoints.up('desktop_1440')]` here for consistency in the code

**Review by YasielCabrera — CHANGES_REQUESTED** [review]

File: src/stories/components/custom-table/row-item/item-core-unit.stories.tsx (hunk 13 above)
Comment (alejandrocabriales) [reply src/stories/components/custom-table/row-item/item-core-unit.stories.tsx:11]: you are right, it I forgot to put there.
done

File: src/stories/containers/cu-table/cu-table.renders.tsx (hunk 14 above)
Comment (alejandrocabriales) [reply src/stories/containers/cu-table/cu-table.renders.tsx]: you are right, done

**Review by YasielCabrera — APPROVED** [review]
LGTM
